README.md

```diff
@@ -1,2 +1,185 @@
-# cert-manager-webhook-ibmcis
-Cert-manager.io webhook to use IBM Cloud Internet Service (ibmcis)
+# IBM Cloud Internet Service Webhook for Cert Manager
+
+This is a webhook solver for [IBM Cloud Internet Service](https://cloud.ibm.com/catalog/services/internet-services#about).
+
+[![Docker Repository on Quay](https://quay.io/repository/borup.work/cert-manager-webhook-ibmcis/status "Docker Repository on Quay")](https://quay.io/repository/borup.work/cert-manager-webhook-ibmcis)
+
+## Prerequisites
+
+* [cert-manager](https://github.com/jetstack/cert-manager): *tested with 1.7.1*
+    - [Installing on Kubernetes](https://cert-manager.io/next-docs/installation/kubernetes/)
+
+```bash
+#kubectl create namespace cert-manager
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.7.1/cert-manager.yaml
+ # kubectl get pods -n cert-manager
+
+```
+
+## Installation
+_Notice: The pod will not startup until the steps under [configuration](#Issuer) is performed (there will be a secret there is not created until the steps are taken, so do not expect it just starts until that is in place )
+
+Assuming your installation has 1. cert-manager running in name-space `cert-manager` and 2. accept this webhook will be installed into namespace `cert-manager-webhook-ibmcis`, 3. the API groups `acme.borup.work` will be used, then it is recommended to install via this pre-defined file .
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/jb-dk/cert-manager-webhook-ibmcis/master/cert-manager-webhook-ibmcis.yaml
+```
+
+### How cert-manager-webhook-ibmcis.yaml is created (information)
+This is just to help me remember how I created the static version of the file and for you to be inspired if you want to try to run it in a different configuration, however I will warn this is not the simplest thing to make it run in a different namespace.
+
+```bash
+helm template --name-template cert-manager-webhook-ibmcis ./deploy/cert-manager-webhook-ibmcis > cert-manager-webhook-ibmcis.yaml
+
+```
+
+### Customized installation
+Only do the steps in this section - customized installation - if you did not do the step in installation.
+```bash
+helm install --name-template cert-manager-webhook-ibmcis ./deploy/cert-manager-webhook-ibmcis
+```
+
+## Issuer
+
+0. (Optional but recommended) Generate a service id (`ibmcloud iam service-id-create cert-manager-webhook-ibmcis-sid -d "Service id that cert-manager-webhook-ibmcis uses"`), grant it "service access" level permission as `reader,writer,manager` to the relevant IBM Cloud Internet Service(s) only (example that grants access to all instances of IBM Cloud Internet Services: `ibmcloud iam service-policy-create cert-manager-webhook-ibmcis-sid --service-name internet-svcs  --roles Reader,Writer,Manager `)
```

Comment:
Could we add more detailed documentation here?
This `service-policy-create` will grant access to all service instances which is not an option in case of bigger organizations which manage multiple products.
We added `--service-instance` here but then discovered, that additional permissions are required in such case.
Full permissions are as follows:
```
Reader, Writer, Manager, Viewer, Operator, Editor, Administrator
```
But I think that we do not need to grant super user permissions here to make it working with the new version of plugin. What are required permissions in such case?

Reply:
update to grant access to specific cis instance.